core/src/main/kotlin/io/customer/sdk/communication/Event.kt

```diff
@@ -3,22 +3,44 @@ package io.customer.sdk.communication
 import java.util.Date
 import java.util.UUID
 
+/**
+ * Base class containing common properties for all events.
+ */
+abstract class BaseEvent(
+    open val storageId: String = UUID.randomUUID().toString(),
+    open val params: Map<String, String> = emptyMap(),
+    open val timestamp: Date = Date()
+)
+
 /**
  * Class to represent an event that can be published and subscribed to using [EventBus].
  */
-sealed class Event {
-    // Unique identifier for the event
-    val storageId: String = UUID.randomUUID().toString()
+sealed class Event : BaseEvent() {
```

Comment:
Why do we need BaseEvent? I think we can have something like this instead

```suggestion
sealed class Event {
    open val storageId: String = UUID.randomUUID().toString(),
    open val params: Map<String, String> = emptyMap(),
    open val timestamp: Date = Date()
```